horovod/mxnet/mpi_ops.cc

```diff
@@ -297,7 +309,21 @@ extern "C" int horovod_mxnet_broadcast_async(NDArray* input, NDArray* output,
                            {output->var()}, FnProperty::kNormal, 0,
                            "HorovodBroadcast");
 #endif
-  return 0;
+  return handle;
+}
+
+extern "C" int horovod_mxnet_poll(int handle) {
+  return handle_manager.PollHandle(handle) ? 1 : 0;
+}
+
+extern "C" int horovod_mxnet_wait_and_clear(int handle) {
+  API_BEGIN();
+  while (!handle_manager.PollHandle(handle)) {
+    std::this_thread::sleep_for(std::chrono::milliseconds(1));
+  }
+  auto status = handle_manager.ReleaseHandle(handle);
+  ThrowIfError(*status);
```

Comment:
do we still need this ThrowIfError here?

Reply:
yes, that's where we actually throw the error from MPI return status.